mapper/src/test/scala/org/scalarelational/mapper/PolymorphSpec.scala

```diff
@@ -0,0 +1,67 @@
+package org.scalarelational.mapper
+
+import org.scalarelational.column.property.{AutoIncrement, NotNull, PrimaryKey}
+import org.scalarelational.h2.{H2Memory, H2Datastore}
+import org.scalarelational.model.Table
+import org.scalatest.{Matchers, WordSpec}
+
+/**
+ * @author Tim Nieradzik <tim@kognit.io>
+ */
+
+class PolymorphSpec extends WordSpec with Matchers {
+  import PolymorphDatastore._
+
+  val insertUsers = Seq(
+    UserGuest("guest"),
+    UserAdmin("admin", true)
+  )
+
+  "Async" should {
+    "create tables" in {
+      session {
+        create(users)
+      }
+    }
+    "insert users" in {
+      session {
+        insertUsers.foreach(users.persist(_).result)
+      }
+    }
+    "query users" in {
+      import users._
+
+      session {
+        val query = select (*) from users
+        val x = query.asCase[User] { row =>
+          if (row(users.isGuest)) classOf[UserGuest]
+          else classOf[UserAdmin]
+        }
+        insertUsers should equal (x.result.converted.toList.map(_.withoutId))
```

Comment:
One problem is that you are explicitly not testing to make sure an id was actually assigned.